src/cmakeTaskProvider.ts

```diff
@@ -551,6 +575,83 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
         }
     }
 
+    private async runPackageTask(): Promise<any> {
+        this.writeEmitter.fire(localize("package.started", "Package task started...") + endOfLine);
+
+        const project: CMakeProject | undefined = this.getActiveProject();
+        if (!project || !await this.isTaskCompatibleWithPresets(project)) {
+            return;
+        }
+        telemetry.logEvent("task", {taskType: "package", useCMakePresets: String(project.useCMakePresets)});
+        await this.correctTargets(project, CommandType.package);
+        const cmakeDriver: CMakeDriver | undefined = (await project?.getCMakeDriverInstance()) || undefined;
+
+        if (cmakeDriver) {
+            let packagePreset: preset.PackagePreset | undefined;
+            this.preset = await this.resolvePresetName(this.preset, project.useCMakePresets, CommandType.package);
+            if (this.preset) {
+                packagePreset = await project?.expandPackagePresetbyName(this.preset);
+                if (!packagePreset) {
+                    log.debug(localize("package.preset.not.found", 'Package preset not found.'));
+                    this.writeEmitter.fire(localize("cpack.failed", "Package preset {0} not found. CPack failed.", this.preset) + endOfLine);
+                    this.closeEmitter.fire(-1);
+                    return;
+                }
+            }
+            const result: number | undefined = cmakeDriver ? await project?.runCPack(cmakeDriver, packagePreset, this) : undefined; //.runCPackCustomized(cmakeDriver, packagePreset, this) : undefined;
+            if (result === undefined) {
+                this.writeEmitter.fire(localize('cpack.run.terminated', 'CPack run was terminated') + endOfLine);
+                this.closeEmitter.fire(-1);
+            } else {
+                this.writeEmitter.fire(localize('cpack.finished', 'CPack finished') + endOfLine);
+                this.closeEmitter.fire(0);
+            }
+        }  else {
+            log.debug(localize("cmake.driver.not.found", 'CMake driver not found.'));
+            this.writeEmitter.fire(localize("cpack.failed", "CPack run failed.") + endOfLine);
+            this.closeEmitter.fire(-1);
+        }
+    }
+
+    // should we have one task per workflow or launch the specific tasks for each entry
```

Comment:
Not sure if I completely understand this, but wouldn't we need a task per workflow so that the workflows can programmatically invoked by other extensions?